src/relay/pass/simplify_inference.cc

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright (c) 2018 by Contributors
+ * \file simplify_inference.cc
+ */
+#include <tvm/relay/pass.h>
+#include <tvm/relay/expr_functor.h>
+#include <tvm/relay/attrs/nn.h>
+#include "./pattern_util.h"
+
+namespace tvm {
+namespace relay {
+
+// TODO: make type generic
+Constant make_const(float x) {
```

Comment:
move to pattern_util.h

```template<typename T>```
Constant MakeConstantScalar(DataType dtype, T value) {
    static_assert(sizeof(T) == dtype.bits() * 8, "size mismatch"); 
}